src/composables/useBarDrag.ts

```diff
@@ -1,108 +1,123 @@
-import { GanttBarObject, GGanttChartPropsRefs } from "./../models/models"
-import useDayjsHelper from "./useDayjsHelper"
+import type { GGanttChartConfig } from "../components/GGanttChart.vue"
+import provideConfig from "../provider/provideConfig"
+import { type Ref, ref } from "vue"
 
+import type { GanttBarObject } from "../types"
+import useDayjsHelper from "./useDayjsHelper"
 import useTimePositionMapping from "./useTimePositionMapping"
-import { Ref, ref } from "vue"
 
-export default function useBarDrag (
+export default function createBarDrag(
```

Comment:
It'd be better to keep the name `useBarDrag`, since it's in accordance with Vue's [naming convention for composables](https://vuejs.org/guide/reusability/composables.html#conventions-and-best-practices) and because the file itself has the same name.

Reply:
The reason I renamed the function was that it isn't meant to be used in places where all the ['Usage Restriction'](https://vuejs.org/guide/reusability/composables.html#conventions-and-best-practices) for composables can be fulfilled. More specific: the useBarDrag / createBarDrag is currently only used and built for the usage inside of initDragOfBar. The latter function is only used inside of `firstMousemoveCallback`

I found this out the hard way because I wanted to inject the config in here as well.
This was not possible because access to an active component instance is necessary.

This led me to the conclusion that we do not have a 'real' composable here.

While writing this I realise that this might also lead to memory leaks like written in that part of the vue docs.

So whatever we conclude from this:
The name of the function and the file should be aligned :D